### PR TITLE
Putting the Colla into our ASCII

### LINKED ISSUE #39: Collaborate over the network

### PR DESCRIPTION
**Pull request by labseven**
Networking actually works!

~~It seemed to be dropping packets somewhere, need to look into that.~~

Closes #39

### CODE REVIEW

**Comment by labseven** [comment]
It's not dropping packets now. Resize caused a sigsev on Evan's machine

**Comment by labseven** [comment]
When a new terminal is opened, the other terminals sometimes do not read from the network socket until they are focused once.

File: src/network.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  fd_set clientfds;
+  int sockfd;
+} Net_cfg;
+
+Canvas *net_init(char *hostname, char *port);
+Net_cfg *net_getcfg();
+void net_handler(View *view);
+void net_send_char(int y, int x, char ch);
+
+#endif
```
Comment (newsch) [inline src/network.h]:
```suggestion
#endif

```
`"files.insertFinalNewline": true` in vscode `config.json`

File: src/server.c [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+#include <signal.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+// #include "network.h"
```
Comment (newsch) [inline src/server.c]: Any reason for this?

**Review by newsch — APPROVED** [review]
Ok, looks good. We can factor out the duplicated code with `frontend.c` later.

**Comment by newsch** [comment]
It works! Pretty fun to use too.

**Pull request merged**